xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -405,5 +410,376 @@ IndexingMap MlirTransposeFusion::GetIndexing(bool input,
   return result;
 }
 
+namespace {
+
+struct HloBytesUsageDesc {
```

Comment:
Can you add comments for all of these members? I don't really understand what they are.